pkg/nsx/services/vpc/vpc.go

```diff
@@ -106,10 +127,15 @@ func InitializeVPC(service common.Service) (*VPCService, error) {
 	wgDone := make(chan bool)
 	fatalErrors := make(chan error)
 
-	wg.Add(2)
-
 	VPCService := &VPCService{Service: service}
-
+	enableAviAllowRule = service.NSXConfig.EnableAviAllowRule && service.NSXClient.NSXCheckVersion(nsx.VpcAviRule)
```

Comment:
please refer to https://github.com/vmware-tanzu/nsx-operator/pull/280/files#diff-c444f711e9191b53952edb65bfd8c644419fc7695c62611dc0fb304b4fb197d6R103
If EnableAviAllowRule is enabled, but NSX version doesn't meet requirement, this is incorrect configuration Since AVI rule feature should only be used in VPC mode and VPC feature supported. 


I think we don't need a flag in VPCService, if really need it, why not add inNSXVersionChecker as we did before for other feature
https://github.com/vmware-tanzu/nsx-operator/blob/vpc_dev/pkg/nsx/client.go#L95

except that we need to check nsx version dynamically for vpc creation or update, however, it looks like not our case here.

Reply:
Done